site/components/pages/Pledge/PledgeForm/index.tsx

```diff
@@ -111,13 +112,34 @@ export const PledgeForm: FC<Props> = (props) => {
   });
   const wallets = useWatch({ name: "wallets", control });
 
+  const checkDuplicateWallets = (wallets: Wallet[]): boolean => {
+    let isDuplicate = false;
+    wallets.map((wallet) => {
+      for (let i = 0; i < wallets.length; i++) {
+        if (
+          wallets[i].address === wallet.address &&
+          wallet.status === "pending" &&
+          wallets[i].status === "pending"
+        ) {
+          isDuplicate = true;
+        }
+      }
+    });
+    return isDuplicate;
+  };
```

Comment:
Does this need to check for duplication of wallet address _and_ status?
Checking for duplicate wallet address seems sufficient to me unless I'm missing something.

Would you find it useful to extend the yup schema similar to the other PR. Something like `yup.array().unique(attribute)?` I can give it a crack.